src/core/shader.js

```diff
@@ -61,18 +62,19 @@ Shader.prototype = {
   },
 
   initVariables: function (data, type) {
-    var self = this;
     var variables = {};
     var schema = this.schema;
     var schemaKeys = Object.keys(schema);
     schemaKeys.forEach(processSchema);
     function processSchema (key) {
       if (schema[key].is !== type) { return; }
       var varType = propertyToThreeMapping[schema[key].type];
-      var varValue = schema[key].parse(data[key] || schema[key].default);
-      variables[key] = {
+      var varKey = key;
+      // For textures, by convention data.src becomes uniform sampler2D map.
```

Comment:
the two existing shaders - standard and flat - both do this goofy src to map translation.  if there are other examples that don't, this can probably go away